pages/index.js

```diff
@@ -47,12 +50,38 @@ export default function HomePage({ onSubmit }) {
 
         setWeather(mainTemp);
         setCondition(condition);
+        if (appUsers) {
+          const user = appUsers.find((user) => user._id === session?.user?.id);
+
+          if (user && user.friends) {
+            setAppUserFriendsList(user.friends);
+          } else {
+            return;
+          }
+        }
```

Comment:
We could also tidy this up a little

``` js
if (appUsers && session?.user?.id) {
  const user = appUsers.find(user => user._id === session.user.id);
  setAppUserFriendsList(user?.friends || []);
}
```

i would also advise setting the default state to be an empty array at first

``` js
const [appUserFriendsList, setAppUserFriendsList] = useState([]);
```